### PR TITLE
fix: harden OTP validation and URI parsing; bubble up TOTP replay controls through otplib

### PR DESCRIPTION
**Pull request by yeojz**
### Summary

  This PR strengthens validation across core OTP flows, tightens OTPAuth URI handling, and wires TOTP replay protection
  (afterTimeStep) through the top-level otplib API.

  ### Key Changes

  - Core validation hardening in @otplib/core:
      - Added CounterNotIntegerError and TimeNotFiniteError.
      - validateCounter now rejects non-integer / non-finite numbers.
      - validateTime now rejects non-finite values.
      - Strengthened createGuardrails with safe-integer checks and invariant checks:
          - MIN_SECRET_BYTES <= MAX_SECRET_BYTES
          - MIN_PERIOD <= MAX_PERIOD
          - bounds validation for all guardrail fields
  - HOTP/TOTP behavior updates:
      - HOTP/TOTP classes no longer require Base32 plugin when secret is already Uint8Array.
  - @otplib/otplib replay-protection wiring:
      - Added afterTimeStep to verify options/types.
      - Passed afterTimeStep through verify and verifySync to TOTP strategy.
  - URI hardening in @otplib/uri:
      - generate now percent-encodes secret query value.
      - parse now requires secret (throws missing-parameter error when absent).
      - Strict integer parsing for counter/period and stricter numeric validation for digits.
  - Test suite updates:
      - Expanded coverage for new validation errors and guardrail invariants.
      - Added replay-protection passthrough tests in otplib.
      - Added URI parsing/encoding edge-case tests.
      - Updated v11/v12 adapter tests for stricter guardrail constraints.

  ### Behavior Notes

  - Parsing URIs without secret now fails instead of returning undefined secret.
  - Invalid guardrail configurations now throw ConfigurationError with explicit messages.

### CODE REVIEW

File: vitest.config.ts [hunk 1]
```diff
@@ -6,14 +6,16 @@ import { defineConfig, defineProject } from "vitest/config";
...
   "@otplib/plugin-crypto-node": path.resolve(__dirname, "packages/plugin-crypto-node/src"),
   "@otplib/plugin-crypto-web": path.resolve(__dirname, "packages/plugin-crypto-web/src"),
   "@otplib/plugin-crypto-noble": path.resolve(__dirname, "packages/plugin-crypto-noble/src"),
+  "@otplib/plugin-base32-alt": path.resolve(__dirname, "packages/plugin-base32-alt/src"),
   "@otplib/plugin-base32-scure": path.resolve(__dirname, "packages/plugin-base32-scure/src"),
+  "@otplib/preset-v11": path.resolve(__dirname, "packages/v11-adapter/src"),
+  "@otplib/v12-adapter": path.resolve(__dirname, "packages/v12-adapter/src"),
   "@repo/testing": path.resolve(__dirname, "internal/testing/src"),
   otplib: path.resolve(__dirname, "packages/otplib/src"),
 };
```
[the remark this answers is not in the document]
Comment (yeojz) [reply vitest.config.ts]: This is intentional for backward compatibility with a published module but retaining consistency in the repository.
